code/src/nuvla/job/actions/nuvlabox_add_ssh_key.py

```diff
@@ -30,25 +28,42 @@ def add_ssh_key(self):
                 break
 
         if credential_id:
-            connector.connect()
-            pubkey = self.job.context[credential_id]['public-key']
-            ssh_keys = connector.nuvlabox.get('ssh-keys', [])
-
-            if credential_id not in ssh_keys:
-                r = connector.nuvlabox_manage_ssh_key('add-ssh-key', pubkey)
+            public_key = self.job.context[credential_id]['public-key']
+            if not self._sanitize_ssh_key(public_key):
+                r = "ssh public key not valid: " + public_key
+                self.job.update_job(status_message=json.dumps(r))
+                return 1
+            self.job.set_progress(10)
+            if os.getenv('KUBERNETES_SERVICE_HOST'):
+                logging.debug('We are using Kubernetes on nuvlabox ID : %s ',nuvlabox_id)
+                connector = K8sSSHKey(api=self.api, nuvlabox_id=nuvlabox_id, job=self.job)
+                connector.handle_ssh_key('add-ssh-key', public_key, credential_id, \
+                    nuvlabox_id=nuvlabox_id)
+            else:
+                connector = NB.NuvlaBox(api=self.api, nuvlabox_id=nuvlabox_id, job=self.job)
+                connector.connect()
+                ssh_keys = connector.nuvlabox.get('ssh-keys', []) 
+                if credential_id not in ssh_keys:
+                    r = connector.nuvlabox_manage_ssh_key('add-ssh-key', public_key)
 
-                update_payload = ssh_keys + [credential_id]
+                    update_payload = ssh_keys + [credential_id]
 
-                connector.commission({"ssh-keys": update_payload})
+                    connector.commission({"ssh-keys": update_payload})
+                else:
+                    r = "Requested SSH key has already been added to the \
+                        NuvlaBox in the past. Nothing to do..."
+                self.job.update_job(status_message=json.dumps(r))
                 self.job.set_progress(100)
-            else:
-                r = "Requested SSH key has already been added to the NuvlaBox in the past. Nothing to do..."
         else:
             raise Exception('Cannot find any reference to an existing credential ID')
 
-        self.job.update_job(status_message=json.dumps(r))
-
         return 0
 
+    def _sanitize_ssh_key(self, public_key):
+        '''Doc string''' 
+        logging.debug("Checking ssh public key %s ",public_key)
+        return re.match(r'^ssh-\w+\s+\S+(\s+\S+)?$', public_key)
+
     def do_work(self):
+        '''Doc string'''
```

Comment:
?

Reply:
Removed the doc string to conform to how other do_work are handled